### PR TITLE
Setup testing and GitHub actions for build on pull request

### PR DESCRIPTION
**Pull request by SimonMTaye**
Setup testing using `vitest` and GitHub actions + Codecov to get test coverage reports as part of pull requests

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
-  "extends": "./node_modules/gts/"
+  "extends": [ 
+    "./node_modules/gts/",
+    "next/core-web-vitals"
+  ],
+  "overrides": [
+    {
+      "files": ["test/**/*.ts", "*.config.ts"],
+      "rules": {
+        "node/no-unpublished-import": ["off"]
```
Comment (SimonMTaye) [inline .eslintrc.json:10]: This rule should only be present for our main code. Test and config code can safely rely on unpublished packages (published means in our `depedencies` part of our `package.json`

File: .eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,14 @@
 {
-  "extends": "./node_modules/gts/"
+  "extends": [ 
+    "./node_modules/gts/",
+    "next/core-web-vitals"
```
Comment (SimonMTaye) [inline .eslintrc.json:4]: add `next.js` eslint rules on top of `gts`

File: .github/workflows/build.yml
Comment (SimonMTaye) [inline .github/workflows/build.yml:1]: Create a GitHub action that builds the code and uploads code coverage report

File: package.json [hunk 3]
```diff
@@ -4,31 +4,36 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "build": "next build",
```
Comment (SimonMTaye) [inline package.json:7]: run linter and tests as part of prebuild stage and run build with `--no-linter` for quicker builds

File: package.json [hunk 4]
```diff
@@ -4,31 +4,36 @@
...
+    "prebuild": "npm run fix && npm run coverage",
+    "build": "next build --no-lint",
     "start": "next start",
-    "lint": "gts lint",
```
Comment (SimonMTaye) [inline package.json:9]: use `eslint` directly instead of `gts` to ensure that we also applies Next's `eslint` rules

File: package.json [hunk 5]
```diff
@@ -4,31 +4,36 @@
...
+    "lint": "eslint */*.ts",
+    "fix": "eslint */*.ts --fix",
     "clean": "gts clean",
-    "compile": "tsc",
-    "fix": "gts fix",
-    "prepare": "npm run compile",
-    "pretest": "npm run compile",
-    "posttest": "npm run lint"
+    "coverage": "vitest run * --coverage",
+    "test": "vitest"
```
Comment (SimonMTaye) [inline package.json:14]: add coverage and testing scripts. `coverage` is meant as part of the build process and will not watch for changes unlike the regular `test` script which is for development

File: package.json [hunk 6]
```diff
@@ -4,31 +4,36 @@
...
   },
   "dependencies": {
+    "firebase": "^10.8.1",
     "next": "14.1.2",
     "react": "^18",
-    "react-dom": "^18",
-    "tailwindcss": "^3.4.1"
```
Comment (SimonMTaye) [inline package.json:21]: tailwind should only be a `devDepedency`. Also added firebase

File: test/basic.example.test.ts
Comment (SimonMTaye) [inline test/basic.example.test.ts:1]: Dummy test to make sure testing scripts were working

File: vitest.config.ts
Comment (SimonMTaye) [inline vitest.config.ts:1]: testing tool configuration

**Review by tandetat — APPROVED** [review]

**Pull request merged**